generator/insn_vdvs2uimmvm.go

```diff
@@ -29,14 +29,33 @@ func (i *Insn) genCodeVdVs2UimmVm(pos int) []string {
 		vs2EMUL1 := LMUL(math.Max(float64(int(c.LMUL)*vs2Size), 1))
 		vs2EEW := c.SEW * SEW(vs2Size)
 
-		vd := int(c.LMUL1)
-		vs2 := 2*int(c.LMUL1) + int(vs2EMUL1)
+		var vd, vs2 int
+		vd = int(c.LMUL1)
+		vs2 = 2*int(c.LMUL1) + int(vs2EMUL1)
+
 		builder.WriteString(i.gWriteRandomData(c.LMUL1))
 		builder.WriteString(i.gLoadDataIntoRegisterGroup(vd, c.LMUL1, SEW(8)))
 
 		builder.WriteString(i.gWriteIntegerTestData(vs2EMUL1, vs2EEW, 1))
 		builder.WriteString(i.gLoadDataIntoRegisterGroup(vs2, vs2EMUL1, vs2EEW))
 
+		if i.Name == "vwsll.vi" {
+			EMUL := int(c.LMUL * 2)
+			if c.LMUL >= 1{
+				for i := EMUL; i <= 31; i += EMUL {
+					vd = i
+					vs2 = int(c.LMUL)
+					break
+				}
+			}else{
+				vd = 4
+				vs2 = 2
+			}
+		}else {
+			vd = int(c.LMUL1)
+			vs2 = 2*int(c.LMUL1) + int(vs2EMUL1)
+		}
```

Comment:
This needs to be more general, please refer to other `vw`-prefixed instruction generators to see how vd and vs2 are computed.